props/prop_base.h

```diff
@@ -543,32 +543,43 @@ class PropBase : CommandParser, Looper, protected SaberBase {
 #endif
 
   // Measure and return the blade identifier resistor.
-  float id() {
+  float RetrieveBladeId() {
     EnableBooster();
     BLADE_ID_CLASS_INTERNAL blade_id;
     float ret = blade_id.id();
-    PVLOG_STATUS << "BLADE ID: " << ret << "\n";
-#ifdef SPEAK_BLADE_ID
-    talkie.Say(spI);
-    talkie.Say(spD);
-    talkie.SayNumber((int)ret);
-#endif
 #ifdef BLADE_DETECT_PIN
     if (!blade_detected_) {
-      STDOUT << "NO ";
       ret += NO_BLADE;
     }
-    STDOUT << "Blade Detected\n";
 #endif
     return ret;
   }
 
+  float PrintAndSpeakId() {
+    float current_id = RetrieveBladeId();
+    PVLOG_STATUS << "BLADE ID: " << current_id << "\n";
+#ifdef BLADE_DETECT_PIN
+    if (!blade_detected_) {
+      STDOUT << "NO ";
+    } else {
+      STDOUT << "Blade Detected\n";
+    }
+#endif
+
+#ifdef SPEAK_BLADE_ID
+    talkie.Say(spI);
+    talkie.Say(spD);
+    talkie.SayNumber((int)current_id);
+#endif
+    return current_id; // not sure why we're returning the ID value here
+  }
+
   size_t FindBestConfig() {
     static_assert(NELEM(blades) > 0, "blades array cannot be empty");
 
     size_t best_config = 0;
     if (NELEM(blades) > 1) {
-      float resistor = id();
+      float resistor = RetrieveBladeId();
```

Comment:
I guess I didn't think this through.
FindBestConfig() needs a parameter (either "quietly" or "announce")
Then we can call id() (or whatever we want to call it) with that parameter, no need for two functions I guess.
(Sorry about the confusion)

Reply:
Alright, how about now?
Calling id(true) will print (and speak) ID, else quiet.
FindBladeAgain() is announced along with the SetPreset routine being run. (except for first 3 seconds of boot to avoid 2 boot.wavs being played)
FindBestConfig() runs quietly otherwise.
"scanid" calls FindBladeAgain(), 
"id" calls id(true) directly, so just an announce.
If BLADE_ID_SCAN_MILLIS is not defined, "scanid" detects the new blade ID.

It's working.
Just that dual boot.wav thing seems it should be addressed.